Code/Framework/AzCore/AzCore/Serialization/ObjectStream.cpp

```diff
@@ -281,7 +281,16 @@ namespace AZ
 
                 if (childClass)
                 {
-                    AZ_Assert(childNode.m_element.m_version <= childClass->m_version, "Serialize was parsing old version class and found newer version element! This should be impossible!");
+                    AZ_Error("Error", childNode.m_element.m_version <= childClass->m_version,
+                        "The class %s supports version %d. Found file class %s with version %d during deserializing, which is a newer version than the code supports. "
```

Comment:
The second string parameter is actually the name of the DataElement that was serialized out.
For a base class it is of the form "BaseClass<n>" where "<n>" is a number.
For a member field, it is the name given to the `SerializeContext::Field` function.
It is not going to be the class name.

Reply:
childClass->m_name  = target class?
childNode.m_element.m_name = name of data element? 
this might make sense to you, but I am not sure the customer would understand what is wrong with these labels
what are the best labels for these parameters so that a customer could find a way to fix their data?